scRNA-seq-advanced/01-read_filter_normalize_scRNA.Rmd

```diff
@@ -79,75 +79,93 @@ ref_dir <- file.path("data", "reference")
   dir.create(normalized_dir, recursive = TRUE)
 }
 
+# output RDS file for normalized data
 output_sce_file <- file.path(normalized_dir, "gliobastoma_normalized_sce.rds")
 ```
 
 
 ## Reading Cell Ranger data
 
 Whether the 10X Cell Ranger data is in Matrix Exchange format or in an HDF5 file, we can use the `read10xCounts()` function from the `DropletUtils` package to read the data into R and create a `SingleCellExperiment` object.
+(If you used something other than Cell Ranger to process the raw data, you would need to use a different function to read it in and create the `SingleCellExperiment` object.)
 
-```{r read SCE}
+```{r read SCE, live=TRUE}
+# read SCE from matrix directory
 raw_sce <- DropletUtils::read10xCounts(raw_matrix_dir) 
 ```
 
-
 Let's look at the contents of the object after reading it in:
 
-```{r view SCE}
+```{r view SCE, live=TRUE}
+# view SCE object
 raw_sce
 ```
 
-We can see that this `SingleCellExperiment` (SCE) object contains 36,601 rows, which correspond to the features (genes) that were analyzed, and 734,492 columns, which correspond to the possible barcode tags that were used in the experiment.
+We can see from this summary that this `SingleCellExperiment` (SCE) object contains 36,601 rows, which correspond to the features (genes) that were analyzed, and 734,492 columns, which correspond to the possible barcode tags that were used in the experiment.
 Note that not all of these barcode tags will have been used, and many of the features may never have been seen either.
 One of our first steps will be to filter out barcodes that were never seen, or that may have only been seen in a droplet that did not contain a cell (an "empty droplet").
 
-### Cell and feature metadata
+### Structure of the `SingleCellExperiment` object
+
+In addition to the main `counts` matrix, which you can see listed as an `assay` in the `SingleCellExperiment` object summary above, the SCE object can contain a number of other tables and matrices, each stored in a "slot" with a particular format. 
+The overall structure of the object can be seen in the figure below, which comes from an [OSCA Introduction chapter](http://bioconductor.org/books/3.16/OSCA.intro/the-singlecellexperiment-class.html).
+
+![structure of a SingleCellExperiment object](diagrams/SingleCellExperiment.png)
 
-In addition to the main `counts` matrix, which you can see listed as an `assay` in the SCE object, there are two other tables of particular interest we will want to look at.
+We have just mentioned the main `assay` slot, which contains full matrices of data (such as transcript counts) with each row a feature and each column a cell. 
+There are also a couple of tables for metadata, and a slot to store reduced-dimension representations of the expression data.
 
-The first is the `rowData` table, which includes the contents of the `features.tsv.gz` file from earlier.
+We'll start with the `rowData` slot, which is a table of metadata for each feature in our object. 
```

Comment:
Might also add "in this case our features are genes" for smoother transition into using "gene" terminology